packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,232 @@
+    this.decoration =
+        const BoxDecoration(color: CupertinoColors.secondarySystemBackground),
+  })  : _type = _CupertinoFormSectionType.padded,
+        assert(children.length > 0),
+        super(key: key);
+
+  final _CupertinoFormSectionType _type;
+
+  /// Sets the form section header. The section header lies above the
+  /// [children] rows.
+  final Widget? header;
+
+  /// Padding for the the [Column] encapsulating the [children].
+  ///
+  /// Defaults to zero padding if constructed with standard
+  /// [CupertinoFormSection] constructor. Defaults to the standard notched-style
+  /// iOS padding when constructing with [CupertinoFormSection.padded].
+  final EdgeInsetsGeometry margins;
+
+  /// The list of rows in the section.
+  ///
+  /// This takes a list, as opposed to a more efficient builder function that
+  /// lazy builds, because forms are intended to be short in row count. It is
+  /// recommended that only [CupertinoFormRow] and [CupertinoTextFormFieldRow]
+  /// widgets be included in the [children] list in order to retain the iOS look.
+  final List<Widget> children;
+
+  /// Sets the decoration for the section.
+  ///
+  /// Defaults to a [CupertinoColors.secondarySystemBackground] background
+  /// color.
+  final BoxDecoration decoration;
+
+  /// Sets the circular border radius for the [Column] encapsulating [children]
+  /// rows.
+  ///
+  /// Defaults to 0.0 when constructed with standard [CupertinoFormSection]
+  /// constructor. Defaults to 10.0 when constructing with
+  /// [CupertinoFormSection.centered].
+  final double borderRadius;
+
+  @override
+  Widget build(BuildContext context) {
+    // Long divider is used for wrapping the top and bottom of rows.
+    // Only used in _CupertinoFormSectionType.base mode
+    final Widget longDivider = Container(
+      color: CupertinoColors.separator,
+      height: 0.5,
+    );
+
+    // Short divider is used between rows.
+    final Widget shortDivider = Container(
+      margin: const EdgeInsetsDirectional.only(start: 15),
+      color: CupertinoColors.separator,
+      height: 0.5,
+    );
+
+    // We construct childrenWithDividers as follows:
+    // Insert a short divider between all rows.
+    // If it is a `_CupertinoFormSectionType.base` type, add a long divider
+    // to the top and bottom of the rows.
+    assert(children.isNotEmpty);
+
+    final List<Widget> childrenWithDividers = <Widget>[];
+
+    if (_type == _CupertinoFormSectionType.base) {
+      childrenWithDividers.add(longDivider);
+    }
+
+    children.sublist(0, children.length - 1).forEach((Widget widget) {
+      childrenWithDividers.add(widget);
+      childrenWithDividers.add(shortDivider);
+    });
+
+    childrenWithDividers.add(children.last);
+    if (_type == _CupertinoFormSectionType.base) {
+      childrenWithDividers.add(longDivider);
+    }
+
+    return DecoratedBox(
+      decoration: const BoxDecoration(
+        color: CupertinoColors.secondarySystemBackground,
```

Comment:
This feels like something that belongs to the form / tableview instead of a section. If the developer wants to use a different background color for the entire tableview (or form) this DeocratedBox will be inconvenient for them as it's not easy to override.

Reply:
This was a mistake on my end. I should've used the `decorator` property but forgot. With decorator, devs can customize the section colors.

I was hesitant to introduce another layer (`CupertinoForm`, for example) just to hold `CupertinoFormSection`s if it doesn't do anything else. Let me know what you think. Also @xster